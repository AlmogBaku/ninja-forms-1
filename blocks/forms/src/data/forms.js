// export const FormFields = (forms, formId) => {
//     return Object.values(forms[formId].formFields)
// }

// export const FormSubmissions = (forms, formId) => {
//     return Object.values(forms[formId].submissions)
// }

// export const SelectedFormFields = (forms, selectedFields, formId) => {
//     return FormFields(forms, formId).filter((field) => {
//         return -1 !== selectedFields.indexOf(field.fieldId)
//     })
// }
export default {
    1: {
        formId: 1,
        formTitle: 'Test Form',
        fields: {
            1: {
                id: 1,
                type: 'textbox',
                label: 'First Name'
            },
            2: {
                id: 2,
                type: 'textbox',
                label: 'Last Name',
            }
        },
        submissions: [
            {
                1: 'Taylor',
                2: 'Swift',
            },
            {
                1: 'Meghan',
                2: 'Trainor',
            },
        ]
    }
}