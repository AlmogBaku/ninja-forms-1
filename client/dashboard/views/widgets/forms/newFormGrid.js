/**
 * Forms Widget Templates View
 *
 * @package Ninja Forms
 * @subpackage Dashboard
 * @copyright (c) 2017 WP Ninjas
 * @since 3.4.25
 */
define( [
    'views/widgets/forms/newFormSection'
], function( SectionView ) {
    var view = Marionette.View.extend( {
        template: "#tmpl-nf-widget-templates-grid",
        className: 'nf-template-grid',
        tagName: 'div',

        initialize: function(){
            var templateGrid = this;
            this.listenTo( nfRadio.channel( 'dashboard' ), 'sort:installedFormTemplates', function( collection ){
                templateGrid.showChildView( 'installed', new SectionView( { collection: collection } ) );
            });
            this.listenTo( nfRadio.channel( 'dashboard' ), 'sort:availableFormTemplates', function( collection ){
                templateGrid.showChildView( 'available', new SectionView( { collection: collection } ) );
            });
        },

        regions: {
            installed: {
                el: '.installed',
                replaceElement: true
            },
            available: {
                el: '.available',
                replaceElement: true
            }
        },

        onRender: function() {
            var collection = nfRadio.channel( 'dashboard' ).request( 'get:formTemplates' );
            if( 'undefined' != typeof collection ) {
                this.showChildView( 'installed', new SectionView( { collection: nfRadio.channel('dashboard').request('get:installedFormTemplates') } ) );
                this.showChildView( 'available', new SectionView( { collection: nfRadio.channel('dashboard').request('get:availableFormTemplates') } ) );
            }
        },
        
    } );
    return view;
} );
