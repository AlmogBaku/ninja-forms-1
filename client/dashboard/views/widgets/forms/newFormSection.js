/**
 * Forms Widget Template Section View
 *
 * @package Ninja Forms
 * @subpackage Dashboard
 * @copyright (c) 2017 WP Ninjas
 * @since 3.4.25
 */
define( [
    'models/formTemplateCollection',
    'views/widgets/forms/newFormTemplate'
], function( TemplateCollection, TemplateView ) {
    var view = Marionette.CollectionView.extend( {
        tagName: 'div',
        className: 'template-list',
        childView: TemplateView,

        initialize: function(){
            this.listenTo( nfRadio.channel( 'widget-forms' ), 'update:filter', this.updateFilter );
        },

        updateFilter: function( term ){
            if ( nfi18n.noResult === this.el.innerHTML ) {
                this.el.innerHTML = '';
            }
            this.setFilter(function (child, index, collection) {
                return 0 <= child.get( 'title' ).toLowerCase().indexOf( term.toLowerCase() );
            });
            if ( this.children.length == 0 ) {
                this.el.innerHTML = nfi18n.noResult;
            }
        }
    } );
    return view;
} );
