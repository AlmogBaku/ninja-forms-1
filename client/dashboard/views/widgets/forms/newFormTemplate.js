/**
 * Forms Widget Individual Form Template View
 *
 * @package Ninja Forms
 * @subpackage Dashboard
 * @copyright (c) 2017 WP Ninjas
 * @since 3.2
 */
define( [], function() {
    var view = Marionette.View.extend( {
        template: "#tmpl-nf-widget-forms-template",

        events: {
        	'click': 'maybeOpenModal'
        },

        /**
         * Prevent page navigation and open our ad modal if this is an ad.
         * @since  3.2.22
         * @param  {object} e Event Object from jQuery
         * @return {void}
         */
        maybeOpenModal: function( e ) {
            e.preventDefault();

            // Settings object for our batch processor
            var settings = {
                // Initial content for the popup modal
                content: '',
                // Batch processor slug. Must match what we have set in our PHP settings array.
                batch_type: 'import_form_template',
                loadingText: 'Importing...',
                extraData: { template: this.model.get( 'id' ) },
                onCompleteCallback: function( response ) {
                    // Bail if we don't return a form ID.
                    if ( 'undefined' == response.form_id ) return false;

                    window.location.href = nfAdmin.builderURL + response.form_id;
                }
            }

            /**
             * Instantiate our batch processor.
             *
             * This will open the modal and present the user with content and buttons.
             */
            new NinjaBatchProcessor( settings );

        	// If this isn't an ad, then early return
        	if ( 'ad' != this.model.get( 'type' ) ) {
        		return true;
        	}
        	// Prevent page navigation.
        	e.preventDefault();

        	// Open our jBox modal
        	var modal = new jBox( 'Modal', {
        		width: 450,
        		title: this.model.get( 'modal-title' ),
        		content: this.model.get( 'modal-content' ),
                closeButton: 'box',
                blockScroll: true
        	} );

        	modal.open();
        }

    } );
    return view;
} );
