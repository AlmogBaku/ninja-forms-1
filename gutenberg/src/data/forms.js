export const FormFields = (forms, formId) => {
    return Object.values(forms[formId].formFields)
}

export const FormSubmissions = (forms, formId) => {
    return Object.values(forms[formId].submissions)
}

export const SelectedFormFields = (forms, selectedFields, formId) => {
    return FormFields(forms, formId).filter((field) => {
        return -1 !== selectedFields.indexOf(field.fieldId)
    })
}