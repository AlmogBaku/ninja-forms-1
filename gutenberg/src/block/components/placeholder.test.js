import forms from '../data/forms'
import Placeholder from './placeholder';
import {
    render, //test renderer
    cleanup, //resets the JSDOM
    fireEvent, //fires events on nodes,
} from "@testing-library/react";

describe("Editor componet", () => {
    afterEach(cleanup);

    it("matches snapshot", () => {
        const props = { inside: <div>Inside Content</div> }
        expect(render(
            <Placeholder {...props} />
        )).toMatchSnapshot();
    });

});