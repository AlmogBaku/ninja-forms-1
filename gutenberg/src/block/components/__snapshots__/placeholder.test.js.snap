// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor componet matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="components-placeholder is-small"
      >
        <iframe
          aria-hidden="true"
          aria-label="resize-listener"
          frameborder="0"
          src="about:blank"
          style="display: block; opacity: 0; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;"
          tabindex="-1"
        />
        <div
          class="components-placeholder__label"
        >
          <div
            size="24"
            style="margin-right: 10px;"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              height="15pt"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              version="1.0"
              viewBox="0 0 624.000000 309.000000"
              width="30pt"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="#000000"
                stroke="none"
                transform="translate(0.000000,309.000000) scale(0.100000,-0.100000)"
              >
                <path
                  d="M5890 2759 c-138 -28 -199 -42 -235 -54 -22 -8 -55 -16 -72 -20 -18
 -4 -35 -11 -38 -16 -4 -5 -14 -9 -24 -9 -36 0 -256 -104 -381 -181 -202 -123
 -441 -334 -563 -496 -18 -23 -37 -43 -43 -43 -6 0 -29 17 -51 38 -246 231
 -460 373 -794 525 -77 36 -196 78 -314 112 -33 9 -75 21 -93 27 -19 5 -57 14
 -85 18 -29 4 -77 13 -107 18 -256 50 -625 49 -880 0 -30 -6 -75 -14 -100 -18
 -53 -9 -130 -27 -185 -45 -22 -8 -55 -16 -72 -20 -18 -4 -35 -11 -38 -16 -4
 -5 -15 -9 -25 -9 -11 0 -34 -6 -52 -14 -18 -8 -73 -31 -122 -51 -500 -207
 -956 -604 -1238 -1077 -58 -99 -178 -338 -178 -356 0 -6 -6 -23 -14 -39 -18
 -35 -59 -148 -76 -208 -8 -28 -48 -191 -72 -295 -16 -72 -29 -214 -35 -377
 l-5 -153 369 0 370 0 17 103 c47 266 178 482 383 629 67 48 326 188 348 188 4
 0 21 6 38 14 18 7 50 20 72 29 22 8 58 22 80 31 49 19 182 57 295 84 544 127
 1099 104 1625 -69 50 -17 99 -34 110 -39 11 -5 49 -21 85 -35 312 -127 497
 -270 628 -486 67 -111 131 -297 132 -381 0 -73 -28 -68 381 -68 l369 0 0 138
 c-1 75 -5 166 -10 202 -5 36 -14 95 -19 132 -16 119 -74 349 -107 426 -8 18
 -14 39 -14 46 0 7 -9 35 -21 62 -11 27 -24 58 -29 69 -21 50 -56 127 -93 204
 -29 61 -37 88 -30 100 12 19 109 50 328 105 188 48 240 60 540 121 253 52 345
 84 345 119 0 20 -103 107 -290 244 -198 146 -190 140 -190 155 0 7 27 34 60
 61 57 47 216 206 270 271 66 78 130 193 130 233 0 20 -30 52 -67 71 -25 13
 -148 13 -213 0z"
                />
                <path
                  d="M1559 320 c-81 -43 -150 -155 -165 -270 l-7 -50 262 0 261 0 -5 28
 c-35 163 -94 261 -183 298 -47 19 -120 17 -163 -6z"
                />
                <path
                  d="M3569 320 c-81 -43 -149 -154 -165 -273 l-6 -47 261 0 261 0 -5 28
 c-35 164 -93 260 -182 297 -48 20 -120 18 -164 -5z"
                />
              </g>
            </svg>
          </div>
          Submissions Table
        </div>
        <div
          class="components-placeholder__instructions"
        >
          Select a form to display the submissions.
        </div>
        <div
          class="components-placeholder__fieldset"
        >
          <div>
            Inside Content
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="components-placeholder is-small"
    >
      <iframe
        aria-hidden="true"
        aria-label="resize-listener"
        frameborder="0"
        src="about:blank"
        style="display: block; opacity: 0; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;"
        tabindex="-1"
      />
      <div
        class="components-placeholder__label"
      >
        <div
          size="24"
          style="margin-right: 10px;"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            height="15pt"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            version="1.0"
            viewBox="0 0 624.000000 309.000000"
            width="30pt"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="#000000"
              stroke="none"
              transform="translate(0.000000,309.000000) scale(0.100000,-0.100000)"
            >
              <path
                d="M5890 2759 c-138 -28 -199 -42 -235 -54 -22 -8 -55 -16 -72 -20 -18
 -4 -35 -11 -38 -16 -4 -5 -14 -9 -24 -9 -36 0 -256 -104 -381 -181 -202 -123
 -441 -334 -563 -496 -18 -23 -37 -43 -43 -43 -6 0 -29 17 -51 38 -246 231
 -460 373 -794 525 -77 36 -196 78 -314 112 -33 9 -75 21 -93 27 -19 5 -57 14
 -85 18 -29 4 -77 13 -107 18 -256 50 -625 49 -880 0 -30 -6 -75 -14 -100 -18
 -53 -9 -130 -27 -185 -45 -22 -8 -55 -16 -72 -20 -18 -4 -35 -11 -38 -16 -4
 -5 -15 -9 -25 -9 -11 0 -34 -6 -52 -14 -18 -8 -73 -31 -122 -51 -500 -207
 -956 -604 -1238 -1077 -58 -99 -178 -338 -178 -356 0 -6 -6 -23 -14 -39 -18
 -35 -59 -148 -76 -208 -8 -28 -48 -191 -72 -295 -16 -72 -29 -214 -35 -377
 l-5 -153 369 0 370 0 17 103 c47 266 178 482 383 629 67 48 326 188 348 188 4
 0 21 6 38 14 18 7 50 20 72 29 22 8 58 22 80 31 49 19 182 57 295 84 544 127
 1099 104 1625 -69 50 -17 99 -34 110 -39 11 -5 49 -21 85 -35 312 -127 497
 -270 628 -486 67 -111 131 -297 132 -381 0 -73 -28 -68 381 -68 l369 0 0 138
 c-1 75 -5 166 -10 202 -5 36 -14 95 -19 132 -16 119 -74 349 -107 426 -8 18
 -14 39 -14 46 0 7 -9 35 -21 62 -11 27 -24 58 -29 69 -21 50 -56 127 -93 204
 -29 61 -37 88 -30 100 12 19 109 50 328 105 188 48 240 60 540 121 253 52 345
 84 345 119 0 20 -103 107 -290 244 -198 146 -190 140 -190 155 0 7 27 34 60
 61 57 47 216 206 270 271 66 78 130 193 130 233 0 20 -30 52 -67 71 -25 13
 -148 13 -213 0z"
              />
              <path
                d="M1559 320 c-81 -43 -150 -155 -165 -270 l-7 -50 262 0 261 0 -5 28
 c-35 163 -94 261 -183 298 -47 19 -120 17 -163 -6z"
              />
              <path
                d="M3569 320 c-81 -43 -149 -154 -165 -273 l-6 -47 261 0 261 0 -5 28
 c-35 164 -93 260 -182 297 -48 20 -120 18 -164 -5z"
              />
            </g>
          </svg>
        </div>
        Submissions Table
      </div>
      <div
        class="components-placeholder__instructions"
      >
        Select a form to display the submissions.
      </div>
      <div
        class="components-placeholder__fieldset"
      >
        <div>
          Inside Content
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
